validator/client/src/main/java/tech/pegasys/artemis/validator/client/duties/AggregationDuty.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.artemis.validator.client.duties;
+
+import com.google.common.primitives.UnsignedLong;
+import java.util.Optional;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import tech.pegasys.artemis.datastructures.operations.AggregateAndProof;
+import tech.pegasys.artemis.datastructures.operations.Attestation;
+import tech.pegasys.artemis.datastructures.operations.AttestationData;
+import tech.pegasys.artemis.util.async.SafeFuture;
+import tech.pegasys.artemis.util.bls.BLSSignature;
+import tech.pegasys.artemis.validator.api.ValidatorApiChannel;
+
+public class AggregationDuty implements Duty {
+  private final ConcurrentMap<Integer, CommitteeAggregator> aggregatorsByCommitteeIndex =
+      new ConcurrentHashMap<>();
+  private final UnsignedLong slot;
+  private final ValidatorApiChannel validatorApiChannel;
+
+  public AggregationDuty(final UnsignedLong slot, final ValidatorApiChannel validatorApiChannel) {
+    this.slot = slot;
+    this.validatorApiChannel = validatorApiChannel;
+  }
+
+  public void addValidator(
+      final int validatorIndex,
+      final BLSSignature proof,
+      final int attestationCommitteeIndex,
+      final SafeFuture<Optional<Attestation>> unsignedAttestationFuture) {
+    aggregatorsByCommitteeIndex.computeIfAbsent(
```

Comment:
Might be worth adding a comment here explaining that we will produce at most 1 aggregation per committee

Reply:
Done.